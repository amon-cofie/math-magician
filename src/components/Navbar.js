import React from 'react';
import { NavLink } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';

const Navbar = () => {
  const links = [
    {
      id: uuidv4(),
      path: '/',
      text: 'Home',
    },
    {
      id: uuidv4(),
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: uuidv4(),
      path: '/quote',
      text: 'Quote',
    },
  ];

  return (
    <nav className="navbar-container">
      <h1 className="main-heading">Math Magicians</h1>
      <ul className="nav-links">
        {links.map((link) => (
          <li data-testid={link.text} key={link.id}>
            <NavLink to={link.path}>{link.text}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
