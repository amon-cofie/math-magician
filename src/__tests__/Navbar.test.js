import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar'

test('renders navbar component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});