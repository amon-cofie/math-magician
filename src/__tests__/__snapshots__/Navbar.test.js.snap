// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar component snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar-container"
  >
    <h1
      class="main-heading"
    >
      Math Magicians
    </h1>
    <ul
      class="nav-links"
    >
      <li
        data-testid="Home"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        data-testid="Calculator"
      >
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li
        data-testid="Quote"
      >
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
